### PR TITLE
CDI-563 Event.fireAsync() - clarify the usage of the returned CompletionStage

### PR DESCRIPTION
**Pull request by antoinesd**
CDI-563 Event.fireAsync() - clarify the usage of the returned CompletionStage

### CODE REVIEW

File: api/src/main/java/javax/enterprise/event/Event.java [hunk 1]
```diff
@@ -123,14 +126,16 @@
...
      * A custom {@link Executor} will be used to make asynchronous calls 
      * </p>
      *
      * @param event the event object
      * @param executor a custom executor to execute asynchronous event
      * @return a {@link CompletionStage} allowing further pipeline composition on the asynchronous operation.
-     *         if any of the synchronous or asynchronous observers notfied by this event throws an exception
+     *         Default asyncrhonous execution facility of this stage will be based on a container default {@link Executor}
+     *         specific to implementation.
+     *         if any of the synchronous or asynchronous observers notified by this event throws an exception
```
Comment (tremes) [inline api/src/main/java/javax/enterprise/event/Event.java]: Typo in "Default asynchronous execution" and mentioning of synchronous observers should be removed.

File: spec/src/main/asciidoc/core/events.asciidoc [hunk 2]
```diff
@@ -168,6 +168,8 @@ If the container is unable to resolve the parameterized type of the event object
 The method `fireAsync()` may be called with a specific `Executor` object to be used to invoke observers method.
 The container should provide a default `Executor` when `fireAsync()` is called without specifying one.
```
Comment (tremes) [inline spec/src/main/asciidoc/core/events.asciidoc]: I would change following sentence 
> The container should provide a default `Executor` ...
to 
> The container must provide a default `Executor` ...
WDYT?

Comment (antoinesd) [inline spec/src/main/asciidoc/core/events.asciidoc]: Well, as it is implementation specific I'm not sure...

Comment (mkouba) [inline spec/src/main/asciidoc/core/events.asciidoc]: +1 for MUST. I think async events must always work even if an `Executor` is not specified. Although the CDI spec does not use key words from [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt) we should probably follow these rules where possible.

File: spec/src/main/asciidoc/core/events.asciidoc [hunk 3]
```diff
@@ -166,7 +166,13 @@ The methods `fire()` and `fireAsync()` fire an event with the specified qualifie
...
 
 The method `fireAsync()` may be called with a specific `Executor` object to be used to invoke observers method.
-The container should provide a default `Executor` when `fireAsync()` is called without specifying one.
+
+The following elements are container specific:
+
+* the default `Executor` used by the container when `fireAsync()` is called without specifying an `Executor`,
+* the `CompletionStage` returned by `fireAsync` methods,
+* all dependent stages of this initial CompletionStage, and
+* the default asynchronous execution facility used for this stages.
```
Comment (tremes) [inline spec/src/main/asciidoc/core/events.asciidoc]: This last point seems to me unnecesarry IMO. It just repeats CompletionStage javadoc.

Comment (mkouba) [inline spec/src/main/asciidoc/core/events.asciidoc]: used for **these** stages

File: api/src/main/java/javax/enterprise/event/Event.java [hunk 4]
```diff
@@ -123,14 +124,15 @@
...
+     * Fires an event asynchronously with the specified qualifiers and notifies asynchronous observers.
      * A custom {@link Executor} will be used to make asynchronous calls 
      * </p>
      *
      * @param event the event object
      * @param executor a custom executor to execute asynchronous event
      * @return a {@link CompletionStage} allowing further pipeline composition on the asynchronous operation.
-     *         if any of the synchronous or asynchronous observers notfied by this event throws an exception
+     *         Default asynchronous execution facility is container specific.
+     *         if any observer notified by this event throws an exception
```
Comment (tremes) [inline api/src/main/java/javax/enterprise/event/Event.java]: Uppercase "If"?. Same for line 115.

File: spec/src/main/asciidoc/core/events.asciidoc (hunk 3 above)
Comment (antoinesd) [inline spec/src/main/asciidoc/core/events.asciidoc]: @tremes I'm mentioning that we'll have a specific (unspecified) execution facility for the dependent stages, how does it repeat the CS Javadoc?

Comment (tremes) [inline spec/src/main/asciidoc/core/events.asciidoc]: It appears to me as it doesn't bring any new information. We are speaking about "default execution facility" which nobody knows what it is (at least it's my feeling) and adittionaly we are saying that it's container specific. 
Anyway it looks like "testable=false" so don't hesitate to ignore me. :)

Comment (mkouba) [inline spec/src/main/asciidoc/core/events.asciidoc]: The truth is if we say the `CompletionStage` is container specific then its *default asynchronous execution facility* is also implicitly container specific.

Comment (mkouba) [inline spec/src/main/asciidoc/core/events.asciidoc]: First three items are ok.

Comment (antoinesd) [inline spec/src/main/asciidoc/core/events.asciidoc]: Ok I remove it.

**Pull request merged**